packages/manager/apps/pci-object-storage/src/utils.ts

```diff
@@ -1,8 +1,31 @@
+export type TParam<T> = {
+  fn: () => Promise<T>;
+  ruleFn: (value: T) => boolean;
+  maxTries?: number;
+  interval?: number;
+  onSuccess?: ({ value, iteration }: { value: T; iteration: number }) => void;
+  onFail?: (iteration: number) => void;
+};
+
+export const poll = async <T>(param: TParam<T>, iteration = 0) => {
+  const [maxTries, interval] = [param.maxTries || 10, param.interval || 5000];
+  if (iteration < maxTries) {
+    const value: T = await param.fn();
+    if (param.ruleFn(value)) {
+      param?.onSuccess({ value, iteration });
```

Comment:
shouldn't those be wrapped in a try catch in order to call onFail is the 'fn' or 'ruleFn' function throws an error ?

Reply:
i thought about it, now i have an extra opinion, yeah it makes sense, and maybe it should be moved to a common package so that it can be reused